lib/stPacket/validation/findDuplicateDPObjectsByIndices.js

```diff
@@ -0,0 +1,65 @@
+const DPObject = require('../../object/DPObject');
+
+/**
+ * @param {DPObject} originalObject
+ * @param {DPObject} objectToCheck
+ * @param {DPContract} dpContract
+ *
+ * @return {boolean}
+ */
+function isDuplicateByIndices(originalObject, objectToCheck, dpContract) {
+  const indicesDefinition = dpContract.getDPObjectSchema(originalObject.getType()).indices;
+
+  return indicesDefinition
+    // For every index definition check if hashes match
+    // accumulating overall boolean result
+    .reduce((accumulator, definition) => {
+      const properties = Object.keys(definition.properties);
+
+      const [originalHash, hashToCheck] = properties
+        .reduce(([originalAcc, toCheckAcc], property) => [
+          `${originalAcc}:${originalObject.get(property)}`,
+          `${toCheckAcc}:${objectToCheck.get(property)}`,
+        ], ['', '']);
+
+      return accumulator || (originalHash === hashToCheck);
+    }, false);
+}
+
+/**
+ * Find duplicate objects by unique indices
+ *
+ * @typedef findDuplicateDPObjectsByIndices
+ *
+ * @param {Object[]} rawDPObjects
+ * @param {DPContract} dpContract
+ *
+ * @return {Object[]}
+ */
+function findDuplicateDPObjectsByIndices(rawDPObjects, dpContract) {
+  // Convert raw objects to DPObject instances
+  const dpObjects = rawDPObjects.map(o => new DPObject(o));
+
+  const duplicateArrays = dpObjects
```

Comment:
1. Group objects by types
2. Filter out types without unique indices
3. Filter out types with only 1 object
4. Compare objects in the groups